bin/add_attachment.py

```diff
@@ -0,0 +1,103 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Used to add attachments to the Anthology.
+Usage:
+
+  add_attachment.py paper_id URL_OR_PATH TYPE
+
+- The ACL ID of the paper (e.g., P17-1012)
+- The path to the attachment (can be a URL)
+- The attachment type (poster, slides, software)
+
+Puts the file in place and modifies the XML.
+"""
+
+import argparse
+import os
+import shutil
+import ssl
+import sys
+import tempfile
+
+from add_revision import maybe_copy
+
+import lxml.etree as ET
+import urllib.request
+
+def main(args):
+
+    if args.path.startswith('http'):
+        _, input_file_path = tempfile.mkstemp()
+        try:
+            print('Downloading file from {}'.format(args.path), file=sys.stderr)
+            with urllib.request.urlopen(args.path) as url, open(input_file_path, mode='wb') as input_file_fh:
+                input_file_fh.write(url.read())
+        except ssl.SSLError:
+            print('An SSL error was encountered in downloading the files.', file=sys.stderr)
+            sys.exit(1)
+    else:
+        input_file_path = args.path
+
+    volume_letter, year, paper_num = [args.paper_id[0], args.paper_id[1:3], args.paper_id[4:]]
+    volume_id = '{}{}'.format(volume_letter, year)
+    ext = args.path.split('.')[-1]
+
+    file_name = f'{args.paper_id}.{args.type.capitalize()}.{ext}'
+    output_dir = os.path.join(args.attachment_root, volume_letter, volume_id)
+
+    # Make sure directory exists
+    if not os.path.exists(output_dir):
+        print('No such directory "{}"'.format(output_dir), file=sys.stderr)
+        sys.exit(1)
+
+    # Copy file
+    dest_path = os.path.join(output_dir, file_name)
+    maybe_copy(input_file_path, dest_path, do=True)
+
+    # Update XML
+    xml_file = os.path.join(os.path.dirname(sys.argv[0]), '..', 'data', 'xml', f'{volume_id}.xml')
+    tree = ET.parse(xml_file)
+    if not tree.getroot().tail: tree.getroot().tail = '\n'
+    for paper in tree.getroot().findall('paper'):
+        if paper.attrib['id'] == paper_num:
+            attachment = ET.Element('attachment')
+            attachment.attrib['type'] = args.type
+            attachment.text = file_name
+            attachment.tail = '\n  '  # newline and one level of indent
+            paper.append(attachment)
+            print('Adding attachment node to XML', file=sys.stderr)
+
+            tree.write(xml_file, encoding="UTF-8", xml_declaration=True)
+            break
+
+    # Clean up
+    if args.path.startswith('http'):
+        os.remove(input_file_path)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('paper_id', help='The Antholgoy paper ID (e.g., P18-1001)')
+    parser.add_argument('path', type=str, help='Path to the attachment (can be URL)')
+    parser.add_argument('type', type=str, choices='poster slides software'.split(), help='Attachment type')
```

Comment:
"slides" is currently not a recognized type; it's "presentation".

I think this only has consequences for how it's displayed and could be changed/extended in:
https://github.com/acl-org/acl-anthology/blob/master/hugo/layouts/partials/attachment_repr.html#L8